presto-cache/src/test/java/com/facebook/presto/cache/alluxio/ByteArraySeekableStream.java

```diff
@@ -46,6 +46,7 @@ public int read()
     public int read(long position, byte[] buffer, int offset, int length)
             throws IOException
     {
+        seek(position);
```

Comment:
I have a slight concern with this but not sure: Typically the stream has a notion of a position that is changed by commands like seek. 

The read call that takes a position argument is typically not supposed to change that "implicit" position. At least that's how it works for say the regular input file stream etc. Can we keep those semantics to avoid surprise ? Maybe by getting the current position and resetting it if necessary ?

Also, how was this need for L49 caught ? Should we add more tests to catch similar such errors ?